src/allotropy/parsers/revvity_kaleido/kaleido_structure.py

```diff
@@ -0,0 +1,369 @@
+                msg="Unable to find results table.",
+            )
+        except AllotropeConversionError:
+            logging.warning(
+                f"Unable to read analysis result '{analysis_parameter}'. Ignoring"
+            )
+            return None
+
+        for column in results:
+            if str(column).startswith("Unnamed"):
+                results = results.drop(columns=column)
+
+        try:
+            a1_str_value = str(results.iloc[0, 0])
+        except IndexError as e:
+            error = f"Unable to find first position of analysis result '{analysis_parameter}'."
+            raise AllotropeConversionError(error) from e
+
+        if try_float_or_none(a1_str_value) is None:
+            return None
+
+        return AnalysisResult(
+            analysis_parameter=analysis_parameter,
+            results={
+                f"{row}{col}": values[col]
+                for row, values in results.iterrows()
+                for col in results.columns
+            },
+        )
+
+    def get_image_feature_name(self) -> str:
+        return self.analysis_parameter
+
+    def get_result(self, well_position: str) -> str:
+        return str(
+            assert_not_none(
+                self.results.get(str(well_position)),
+                msg=f"Unable to get well at position '{well_position}' from analysis result '{self.analysis_parameter}'.",
+            )
+        )
+
+    def get_image_feature_result(self, well_position: str) -> float:
+        return try_float(
+            self.get_result(well_position),
+            f"analysis result '{self.analysis_parameter}' at '{well_position}'",
+        )
+
+
+@dataclass(frozen=True)
+class MeasurementInfo:
+    instrument_serial_number: str
+    measurement_time: str
+    protocol_signature: str
+    measurement_signature: str
+
+    @staticmethod
+    def get_instrument_serial_number(elements: dict[str, str]) -> str:
+        return assert_not_none(
+            elements.get("Instrument Serial Number"),
+            msg="Unable to find Instrument Serial Number in Measurement Information section.",
+        )
+
+    @staticmethod
+    def get_measurement_time(elements: dict[str, str]) -> str:
+        return assert_not_none(
+            elements.get("Measurement Started"),
+            msg="Unable to find Measurement time in Measurement Information section.",
+        )
+
+    @staticmethod
+    def get_protocol_signature(elements: dict[str, str]) -> str:
+        return assert_not_none(
+            elements.get("Protocol Signature"),
+            msg="Unable to find Protocol Signature in Measurement Information section.",
+        )
+
+    @staticmethod
+    def get_measurement_signature(elements: dict[str, str]) -> str:
+        return assert_not_none(
+            elements.get("Measurement Signature"),
+            msg="Unable to find Measurement Signature in Measurement Information section.",
+        )
+
+
+@dataclass(frozen=True)
+class PlateType:
```

Comment:
This is not used. It is very confusing code. 

From what I can tell, you are using this create method to move past the plate type section of the table.

This should not be done this way. You should move the reader (call the code of this create method) inside the `create_v2_data` method.  I know that means that you can't create the other sections inside the constructor, but that little nice bit of formatting is not worth creating a dummy dataclass that does nothing.

The only reason that it could be argued to keep this is if it is going to be used in the future. In this case, you should add a comment explaining this.

Otherwise, it should be removed.

Reply:
The idea of this class was to be used in case some parts of that information were relevant in the future. For the moment I remove that class.